docs/concepts/caching.rst

```diff
@@ -0,0 +1,489 @@
+        """Get the current local temperature from a public API"""
+        return weather_api.Client(**credentials).get_temperature(...)
+
+    def weather_message(
+        current_temperature: float, current_wind: dict, is_rain: bool,
+    ) -> str:
+        """Interpret the temperature and return a message"""
+        if ...:
+            return "what a nice day 🌞"
+        elif ...:
+            return "you should bring your umbrella ☔"
+        else:
+            return "it's starting to feel cold 🥶"
+
+
+Don't fingerprint
+~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :widths: auto
+   :align: center
+
+   * - Try to retrieve result
+     - Store result
+     - Fingerprint result
+     - Store fingerprint
+   * - ✅
+     - ✅
+     - ❌
+     - ❌
+
+Large and complex data objects (e.g., dataframes, machine learning models) can be expensive or unreliable to fingerprint, but may be valuable to cache nonetheless.
+
+In that case, using ``caching_behavior="dont_fingerprint"`` will store the result with a constant fingerprint making the result only retrievable via ``resume_from="latest"``.
+
+For instance, we don't want to store the ``weather_message`` because it's only a set of ``if/else`` that are trivial to compute.
+
+.. code-block:: python
+
+    import weather_api
+    from sklearn.ensemble import HistGradientBoostingRegressor
+
+    def weather_client(credentials: dict) -> weather_api.Client:
+        return weather_api.Client(**credentials)
+
+    def current_temperature(weather_client: weather_api.Client) -> float:
+        """Get the current local temperature from a public API"""
+        return weather_api.Client(**credentials).get_temperature(...)
+
+    @tag(caching_behavior="dont_fingerprint")
+    def weather_predictor(
+        current_temperature: float, current_wind: dict, is_rain: bool,
+    ) -> str:
+        """Train a model to predict the probability of rain"""
+        model = HistGradientBoostingRegressor(...)
+        model.train(...)
+        return model
+
+.. note::
+
+     An alternative solution is to register a custom fingerprinting function by ``type`` (see **Fingerprinting**).
+
+
+Don't store result
+~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :widths: auto
+   :align: center
+
+   * - Try to retrieve result
+     - Store result
+     - Fingerprint result
+     - Store fingerprint
+   * - ✅
+     - ❌
+     - ✅
+     - ❌
+
+If a node produces results that are large in size, not picklable, or very inexpensive to compute, it might be preferrable to not store it. Also, your environment may have storage or I/O limitations (e.g., web service).
+
+
+Specifying ``caching_behavior="dont_store_result"`` means Hamilton won't add the result to the result store, but the fingerprint won't be added to the metadata store since it would point to nothing (i.e., "result store miss"). The fingerprint is only used to create the key for children nodes (e.g., ``recommended_outfit``).
```

Comment:
```suggestion
Specifying ``caching_behavior="dont_store_result"`` means Hamilton won't add the result to the result store, and the fingerprint won't be added to the metadata store since it would point to nothing (i.e., "result store miss"). The fingerprint is only used to create the key for downstream nodes (e.g., ``recommended_outfit``).
```